test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
         ], $classInfo->getTraitAliases());
     }
 
+    public function testGetTraitNamesWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitNames();
+    }
+
+    public function testGetTraitsWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraits();
+    }
```

Comment:
these tests show, that the pre-PR existing API always throws and therefore does not allow to get the interface names, when the class-symbols are not autoloadable.